### PR TITLE
Add byteA API

### PR DESCRIPTION
**Pull request by danog**

### CODE REVIEW

File: src/PqHandle.php [hunk 1]
```diff
@@ -294,6 +295,14 @@ private function release(): void
...
     }
 
+    private function escapeParams(array &$params): void
+    {
+        foreach ($params as &$param) {
+            if ($param instanceof ByteA) {
+                $param = $this->handle->escapeByteA($param->getString());
+            }
+        }
+    }
```
Comment (kelunik) [inline src/PqHandle.php:305]: This should return a new array instead of using references.

Comment (danog) [reply src/PqHandle.php:306]: Done!

File: src/functions.php [hunk 2]
```diff
@@ -88,6 +88,9 @@ function cast($value)
 
         case "object":
             if (!\method_exists($value, "__toString")) {
+                if ($value instanceof ByteA) {
```
Comment (trowski) [inline src/functions.php:91]: This shouldn't be here, as a `ByteA` instance should either have already been converted to a string or this should error.

File: src/functions.php [hunk 3]
```diff
@@ -116,6 +119,9 @@ function encode(array $array): string
 
             case "object":
                 if (!\method_exists($value, "__toString")) {
+                    if ($value instanceof ByteA) {
```
Comment (trowski) [inline src/functions.php:122]: Same here.

File: src/PqHandle.php [hunk 4]
```diff
@@ -294,6 +295,15 @@ private function release(): void
         $deferred->resolve();
     }
 
+    private function escapeParams(array $params): array
+    {
+        foreach ($params as $key => $param) {
+            if ($param instanceof ByteA) {
```
Comment (trowski) [inline src/PqHandle.php:301]: This needs to be recursive if the param is an array.

**Review by trowski — COMMENTED** [review]
~~Result sets with ByteA columns should also return `ByteA` instances in the result set.~~

Turns out that is more difficult than I thought. I've never used ByteA columns, is the string usually escaped in the result set? I assume it is since functions like [pg_unescape_bytea](https://www.php.net/manual/en/function.pg-unescape-bytea.php) exist, but some of the documentation I've read sounds like you can have the server automatically unescape the value in a result set, is that correct?

**Pull request closed without merging**